crates/db_views_moderator/src/modlog_combined_view.rs

```diff
@@ -0,0 +1,1377 @@
+      )
+      .or(
+        mod_remove_post::id
+          .is_not_null()
+          .and(post::creator_id.eq(other_person)),
+      )
+      .or(mod_transfer_community::other_person_id.eq(other_person));
+
+    let comment_join = mod_remove_comment::comment_id.eq(comment::id);
+
+    let post_join = admin_purge_comment::post_id
+      .eq(post::id)
+      .or(mod_feature_post::post_id.eq(post::id))
+      .or(mod_lock_post::post_id.eq(post::id))
+      .or(
+        mod_remove_comment::id
+          .is_not_null()
+          .and(comment::post_id.eq(post::id)),
+      )
+      .or(mod_remove_post::post_id.eq(post::id));
+
+    let community_join = admin_purge_post::community_id
+      .eq(community::id)
+      .or(mod_add_community::community_id.eq(community::id))
+      .or(mod_ban_from_community::community_id.eq(community::id))
+      .or(
+        mod_feature_post::id
+          .is_not_null()
+          .and(post::community_id.eq(community::id)),
+      )
+      .or(mod_hide_community::community_id.eq(community::id))
+      .or(
+        mod_lock_post::id
+          .is_not_null()
+          .and(post::community_id.eq(community::id)),
+      )
+      .or(
+        mod_remove_comment::id
+          .is_not_null()
+          .and(post::community_id.eq(community::id)),
+      )
+      .or(mod_remove_community::community_id.eq(community::id))
+      .or(
+        mod_remove_post::id
+          .is_not_null()
+          .and(post::community_id.eq(community::id)),
+      )
+      .or(mod_transfer_community::community_id.eq(community::id));
+
+    let instance_join = admin_allow_instance::instance_id
+      .eq(instance::id)
+      .or(admin_block_instance::instance_id.eq(instance::id));
+
+    let mut query = modlog_combined::table
+      .left_join(admin_allow_instance::table)
+      .left_join(admin_block_instance::table)
+      .left_join(admin_purge_comment::table)
+      .left_join(admin_purge_community::table)
+      .left_join(admin_purge_person::table)
+      .left_join(admin_purge_post::table)
+      .left_join(mod_add::table)
+      .left_join(mod_add_community::table)
+      .left_join(mod_ban::table)
+      .left_join(mod_ban_from_community::table)
+      .left_join(mod_feature_post::table)
+      .left_join(mod_hide_community::table)
+      .left_join(mod_lock_post::table)
+      .left_join(mod_remove_comment::table)
+      .left_join(mod_remove_community::table)
+      .left_join(mod_remove_post::table)
+      .left_join(mod_transfer_community::table)
+      // The moderator
+      .left_join(person::table.on(moderator_names_join))
+      // The comment
+      .left_join(comment::table.on(comment_join))
+      // The post
+      .left_join(post::table.on(post_join))
+      // The community
+      .left_join(community::table.on(community_join))
+      // The instance
+      .left_join(instance::table.on(instance_join))
+      // The other / modded person
+      .left_join(aliases::person1.on(other_person_join))
```

Comment:
The id for these should be columns in the combined table, and those columns should be indexed

Reply:
The combined tables don't hold all of the possible columns in every one of their joins, that could get wildly complicated very quickly. It only holds a sorting column, and the specific ids of the tables they're joined to. 

If that needs to be reworked later, then we can revisit, but I could see it getting overwhemingly complicated for both the combined tables and the triggers needed to fill them.